### PR TITLE
Fix vulnerabilities in root of extension

### PR DESCRIPTION
**Pull request by KonstantinTyukalov**
**Task name**: App store vsts extension

**Description**:
- Fixed minor vulnerabilities by running `npm audit fix` command
- Updated tfx-cli to the latest version
- Fixed last vulnerabilities by running `npm audit fix --force` command

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected - it needs to provide smoke tests for it

### CODE REVIEW

**Review by SvetlanaMaliugina — APPROVED** [review]
Added changes:
- changed ```mocha``` version and  ```make.js``` and ```make-util.js``` files to run tests correctly. 
- changed ```azure-pipelines-task-lib``` version, because it uses correct ```shelljs``` version(0.8.5)
- run tests with all changes - passed successfully

**Review by vmapetr — APPROVED** [review]

File: package-lock.json [hunk 1]
```diff
@@ -97,22 +97,78 @@
       "integrity": "sha512-eqz8c/0kwNi/OEHQfvIuJVLTst3in0e7uTKeuY+WL/zfKn0xVujOTp42bS/vUUokhK5P2BppLd9JXMOMHcgbjA==",
       "dev": true
     },
+    "@ungap/promise-all-settled": {
```
Comment (alexander-smolyakov) [inline package-lock.json:100]: Please update the version as well

https://github.com/microsoft/app-store-vsts-extension/blob/fbaa81b7d198fa5156c44ca87bf332850d71b8a9/package-lock.json#L2-L4

File: make-util.js [hunk 2]
```diff
@@ -1144,7 +1144,7 @@ exports.toOverrideString = toOverrideString;
 
 var createExtension = function(manifest) {
     ensureTool('tsc', '--version', 'Version 3.2.2');
-    ensureTool('mocha', '--version', '5.2.0');
+    ensureTool('mocha', '--version', '9.2.2');
```
Comment (alexander-smolyakov) [inline make-util.js:1147]: Could it make sense to extract the version as const? (e.g. `TARGET_MOCHA_VERSION`)

**Review by alexander-smolyakov — COMMENTED** [review]
Please take a look at the comments below

**Comment by max-zaytsev** [comment]
Tested uploading to on-prem server
image

File: package-lock.json (hunk 1 above)
Comment (max-zaytsev) [reply package-lock.json]: Thanks! Fixed

File: dev-dependencies-constants.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+exports.MOCHA_TARGET_VERSION = '6.2.3';
+exports.TSC_MIN_VERSION = '3.2.2';
+exports.NODE_MIN_VERSION = '10.0.0';
+exports.NPM_MIN_VERSION = '5.6.0';
```
Comment (alexander-smolyakov) [inline dev-dependencies-constants.js:4]: nit: EOL

File: dev-dependencies-constants.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+exports.MOCHA_TARGET_VERSION = '6.2.3';
```
Comment (alexander-smolyakov) [inline dev-dependencies-constants.js:1]: Could you please add JSdoc comments to the file header?

File: make-util.js [hunk 5]
```diff
@@ -19,7 +19,13 @@ var makeOptions = require('./make-options.json');
...
 var cultureNames = [ 'cs', 'de', 'es', 'fr', 'it', 'ja', 'ko', 'pl', 'pt-BR', 'ru', 'tr', 'zh-Hans', 'zh-Hant' ];
 
-var allowedTypescriptVersions = ['2.3.4', '4.0.2'];
+// core dev-dependencies constants
+const constants = require('./dev-dependencies-constants');
+
+const MOCHA_TARGET_VERSION = constants.MOCHA_TARGET_VERSION;
+const TSC_MIN_VERSION = constants.TSC_MIN_VERSION;
+
+var allowedTypescriptVersions = [TSC_MIN_VERSION, '4.0.2'];
```
Comment (alexander-smolyakov) [inline make-util.js:28]:
```suggestion
const allowedTypescriptVersions = [TSC_MIN_VERSION, TSC_CURRENT_VERSION];
```

Could you please move `'4.0.2'` to consts?

File: make.js [hunk 6]
```diff
@@ -62,10 +62,16 @@ var packagePath = path.join(__dirname, '_package');
...
+// core dev-dependencies constants
+const constants = require('./dev-dependencies-constants');
+
+const MOCHA_TARGET_VERSION = constants.MOCHA_TARGET_VERSION;
+const TSC_MIN_VERSION = constants.TSC_MIN_VERSION;
+const NODE_MIN_VERSION = constants.NODE_MIN_VERSION;
+const NPM_MIN_VERSION = constants.NPM_MIN_VERSION;
+
+if (semver.lt(process.versions.node,  NODE_MIN_VERSION)) {
+    fail('requires node >= ' +  NODE_MIN_VERSION + '.  installed: ' + process.versions.node);
```
Comment (alexander-smolyakov) [inline make.js:74]:
```suggestion
    fail(`requires node >= ${NODE_MIN_VERSION}. installed: ${process.versions.node}`);
```

**Review by alexander-smolyakov — APPROVED** [review]
LGTM and please take a look at the comments below

**Pull request closed without merging**